### ISSUE TITLE
Support Blender 2.7

### ISSUE THREAD

**Issue by ProjectBarks** [AnalyticalGraphicsInc/ion-blender-exporter#5]
## The Problem
Currently the project is built for blender 2.8 which uses python 3 instead of version 2. It also means the glTF exporter is not bundled by default. If the demand is high enough or if there is enough residual users of blender 2.79 that could warrant making a 2.79 version.

## Execution
To do this we would have to backport python 3 code to version 2 using a library like [3to2](https://pypi.org/project/3to2/). Additionally, the [gltf exporter](https://github.com/KhronosGroup/glTF-Blender-IO) would have to be bundled with our project so the user has the library by default. As long as the library for converting from python 3 to python 2 works the bundling would not be too hard. 

## Final Thoughts
I think version 2.8 will suffice for but it could be something we decide to approach if the demand is there.

**Comment by emackey** [comment]
The bigger issue is the Blender Python API differences between 2.79 and 2.80.  A number of features were added, removed, and renamed.

The glTF-Blender-IO project is written to support both versions directly (via `if` statements), but the Blender maintainers insist on a "clean" 2.80-only version to bundle.  So we have a script called [convert_to_strict_2.80.py](https://github.com/KhronosGroup/glTF-Blender-IO/blob/master/tools/convert_to_strict_2.80.py) that looks for these specially-formatted `if bpy.app.version < (2, 80, 0):` statements that maintain cross-version compatibility, and strips them out, leaving only the `else` clause that is 2.80-specific.

So, the raw code in the GitHub repo works in both 2.79 and 2.80, but produces a number of warnings in 2.80.  The converted code has the 2.79 stuff stripped out, and ships as a 2.80-only addon with no warnings.

**Comment by emackey** [comment]
2.80 stable is due in July, so this may all be a moot point pretty soon.

Having the repo's raw code work directly in 2.80 is a great help for interactive debugging too.

**Comment by ProjectBarks** [comment]
@emackey My biggest concern is people will want to stick with 2.79 since the UI has been largely changed. Yeah worse case scenario I will try and make a python addon that converts 2.8 code to 2.79 on top of the python 3 to 2.7 converter.

**Comment by emackey** [comment]
What I'm suggesting is, rather than invent a new conversion script from scratch, you should take the time to understand what glTF-Blender-IO is doing, and consider doing exactly the same thing, with the same conversion script that already exists.  Especially since glTF-Blender-IO is your primary dependency, it makes sense for the structure and conversions to be the same.  You could probably also borrow much of the test framework as well, but that's another topic.

**Comment by ProjectBarks** [comment]
Yeah I haven't put too much thought into the approach I want to take. My first thought is just to do a find and replace for the renamed components but now I am remembering I am using some 2.8 only features ie timers.

**Comment by mramato** [comment]
I agree that we will most likely never support 2.79, but we can leave this open for the time being so people can +1 it if they would like to see it happen.  We definitely shouldn't spend any time on it otherwise.

**Comment by Ramirez8052** [comment]
Hello, 
I'm not a developer but a new user of Blender Cesium. 
I would like to use this addon on a regular basis, as it will be very useful in my work. But I have 2 problems. 
1. It doesn't work with Blender version 3.3. 
2. I did the exercise with version 2.80 as indicated on the Cesium website, and it works. However, I can't get my asset in the right place. 
In Cesium ion, when importing, it asks you to set the location, which I did, but it still puts my object in the center of the globe. 
Can you help me?
